wgpu-hal/src/vulkan/adapter.rs

```diff
@@ -669,8 +669,20 @@ impl PhysicalDeviceCapabilities {
         }
 
         // Optional `VK_EXT_robustness2`
+        // Intel iGPUs with outdated drivers can break rendering if `VK_EXT_robustness2` is pushed.
+        // Driver version 31.0.101.2115 works, but there's probably an earlier functional version.
         if self.supports_extension(vk::ExtRobustness2Fn::name()) {
-            extensions.push(vk::ExtRobustness2Fn::name());
+            const DRIVER_VERSION_INTEL_31_0_101: u32 = 0x194000;
+            const DRIVER_VERSION_INTEL_WORKING: u32 = DRIVER_VERSION_INTEL_31_0_101 + 2115;
+
+            let props = self.properties;
+            let is_intel_igpu_outdated = props.vendor_id == crate::auxil::db::intel::VENDOR
+                && props.device_type == vk::PhysicalDeviceType::INTEGRATED_GPU
+                && props.driver_version < DRIVER_VERSION_INTEL_WORKING;
+
+            if !is_intel_igpu_outdated {
```

Comment:
Could we put a `log::trace` level log when this condition isn't hit - with something like "Disabling robustBufferAccess2: ntegrated Intel Driver found with driver version {} which is less than known good driver 31.0.101.2115"

Also we should make sure this doesn't hit mesa drivers as well, only proprietary.

Reply:
What do you mean by "this"? The issue or the `is_intel_igpu_outdated`? If the latter, doesn't MESA have its own vendor id here https://github.com/gfx-rs/wgpu/blob/43b74f9892abab4069e653671f75b3ff5850aa62/wgpu-hal/src/auxil/mod.rs#L28-L34 or is that what the "Mesa does not actually have a PCI vendor id." refers to, where it could be Intel's `0x8086`

Reply:
In Mesa, official drivers are considered to have the vendor id of the hardware vendor. Community drivers such as asahi use Mesa's vendor id.

Reply:
Ok I think my newest addition to the conditional accounts for mesa drivers now